### PR TITLE
For #35: removed redundant methods for YamlSettings

### PR DESCRIPTION
**Pull request by paulodamaso**
For #35: removed redundant methods for YamlSettings.

- Removed redundant methods from `YamlSetting`, which are already executed in `StrictYamlMapping`;
- Left question in issue about how `StrictYamlMapping` and `YamlMapping` should behave

### CODE REVIEW

File: src/main/java/com/artipie/YamlSettings.java [hunk 1]
```diff
@@ -53,61 +53,12 @@ public YamlSettings(final String content) {
...
     @Override
     public Storage storage() throws IOException {
-        final YamlMapping yaml = this.storageYaml();
-        final String type = string(yaml, "type");
-        if (type.equals("fs")) {
-            return new FileStorage(Path.of(string(yaml, "path")));
+        final YamlMapping yaml =
+            Yaml.createYamlInput(this.content).readYamlMapping().yamlMapping("meta").yamlMapping("storage");
+        final String type =  yaml.string("type");
+        if ("fs".equals(yaml.string("type"))) {
```
Comment (HDouss) [inline src/main/java/com/artipie/YamlSettings.java]: @paulodamaso Please use the `type` variable declared in L58

**Comment by 0crat** [comment]
This pull request [#39](https://github.com/artipie/artipie/issues/39) is assigned to @hdouss[/z](https://www.0crat.com/u/hdouss), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/c544f57f-5136-471b-93ef-65d883e1e75b); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by paulodamaso** [comment]
@HDouss Thank you, I'll fix those errors when I finish the PR

**Comment by paulodamaso** [comment]
@HDouss Please review again, now the PR is ready.

**Comment by HDouss** [comment]
@g4s8 Please review and/or merge

File: src/main/java/com/artipie/YamlSettings.java [hunk 2]
```diff
@@ -53,61 +54,17 @@ public YamlSettings(final String content) {
...
-    private static String string(final YamlMapping yaml, final String key) {
-        final String string = yaml.string(key);
-        if (string == null) {
-            throw new IllegalArgumentException(
-                String.format("Cannot find '%s' string:\n%s", key, yaml)
+        final YamlMapping yaml =
+            new StrictYamlMapping(
+                Yaml.createYamlInput(this.content)
+                .readYamlMapping()
+                .yamlMapping("meta")
```
Comment (g4s8) [inline src/main/java/com/artipie/YamlSettings.java]: @paulodamaso previously we checked that the first and second mappings are not nulls, now we check only last string using  `StrictYamlMapping`, so if `meta` mapping is `null` then we'll get `NullPointerException` here. Maybe we need some unit tests before refactoring?

Comment (paulodamaso) [reply src/main/java/com/artipie/YamlSettings.java]: @g4s8 This case is already being tested in `shouldFailProvideStorageFromBadYaml`, first and second test cases

**Comment by paulodamaso** [comment]
@g4s8 PLease take a look, we already test these conditions in `shouldFailProvideStorageFromBadYaml`

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR, performer @hdouss[/z](https://www.0crat.com/u/hdouss)[/z](https://www.0crat.com/u/hdouss) didn't make CR comments

**Comment by 0crat** [comment]
Job was finished in 29 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Pull request merged**